classes/03class/exercises/c03-aws02/zarajoy/README.md

```diff
@@ -0,0 +1,242 @@
+                    "LoadBalancerAddresses": []
+                }
+            ],
+            "SecurityGroups": [
+                "sg-0b91305354d705dbb"
+            ],
+            "IpAddressType": "ipv4"
+        }
+    ]
+}
+#check load balancer is active
+z@bacon:~$ aws elbv2 describe-load-balancers help
+z@bacon:~$ aws elbv2 describe-load-balancers --load-balancer-arns arn:aws:elasticloadbalancing:ap-southeast-2:512742231244:loadbalancer/app/c03-alb/6a5a70ee5d6a2632
+{
+    "LoadBalancers": [
+        {
+            "LoadBalancerArn": "arn:aws:elasticloadbalancing:ap-southeast-2:512742231244:loadbalancer/app/c03-alb/6a5a70ee5d6a2632",
+            "DNSName": "c03-alb-767053992.ap-southeast-2.elb.amazonaws.com",
+            "CanonicalHostedZoneId": "Z1GM3OXH4ZPM65",
+            "CreatedTime": "2020-07-11T02:42:31.280Z",
+            "LoadBalancerName": "c03-alb",
+            "Scheme": "internet-facing",
+            "VpcId": "vpc-09d2f2719d50d1f7f",
+            "State": {
+                "Code": "active"
+            },
+            "Type": "application",
+            "AvailabilityZones": [
+                {
+                    "ZoneName": "ap-southeast-2a",
+                    "SubnetId": "subnet-0907f22726c996fd5",
+                    "LoadBalancerAddresses": []
+                },
+                {
+                    "ZoneName": "ap-southeast-2b",
+                    "SubnetId": "subnet-0abbcb057289b670d",
+                    "LoadBalancerAddresses": []
+                }
+            ],
+            "SecurityGroups": [
+                "sg-0b91305354d705dbb"
+            ],
+            "IpAddressType": "ipv4"
+        }
+    ]
+}
+#create target group
+z@bacon:~$ aws elbv2 create-target-group --name c03-alb-target --protocol HTTP --port 80 --vpc-id vpc-09d2f2719d50d1f7f
+{
+    "TargetGroups": [
+        {
+            "TargetGroupArn": "arn:aws:elasticloadbalancing:ap-southeast-2:512742231244:targetgroup/c03-alb-target/b42f166a2bbac7a2",
+            "TargetGroupName": "c03-alb-target",
+            "Protocol": "HTTP",
+            "Port": 80,
+            "VpcId": "vpc-09d2f2719d50d1f7f",
+            "HealthCheckProtocol": "HTTP",
+            "HealthCheckPort": "traffic-port",
+            "HealthCheckEnabled": true,
+            "HealthCheckIntervalSeconds": 30,
+            "HealthCheckTimeoutSeconds": 5,
+            "HealthyThresholdCount": 5,
+            "UnhealthyThresholdCount": 2,
+            "HealthCheckPath": "/",
+            "Matcher": {
+                "HttpCode": "200"
+            },
+            "TargetType": "instance"
+        }
+    ]
+}
+#Use the register-targets command to register your instances with your target group
+#instances created in c03-aws01
+z@bacon:~$ aws elbv2 register-targets --target-group-arn arn:aws:elasticloadbalancing:ap-southeast-2:512742231244:targetgroup/c03-alb-target/b42f166a2bbac7a2 --targets Id=i-051bfa1e0f24b4f91 Id=i-0e5988470d6ad765f
```

Comment:
What happens if one of your instances goes down?

Probably the ASG will scale up another one to reach the desired state. But will it be registered and reachable via 
load balancer? (No)

For auto (de)registering instances created by the ASG you need to associate the ASG with the Target group:

```
aws autoscaling attach-load-balancer-target-groups \
    --auto-scaling-group-name devops-academy-asg \
    --target-group-arns arn:aws:elasticloadbalancing:ap-southeast-2:149613515908:targetgroup/devops-academy-tg/16871e3110304536
```

Then you don't need to register specific instance IDs in your load balancer